test/travis/check_all_examples_use_standard_init.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+FBASE_SNIPPET=test/travis/firebase_init.cc.snip
+FBASE_NOWIFI_SNIPPET=test/travis/no_firebase_init.cc.snip
+for example in `find examples/ -name *.ino`;
+do
+  echo $example;
+  (xxd -p $example | tr -d '\n' | grep -q `xxd -p $FBASE_SNIPPET | tr -d '\n'`) ||
+  (xxd -p $example | tr -d '\n' | grep -q `xxd -p $FBASE_NOWIFI_SNIPPET | tr -d '\n'`) ||
```

Comment:
Ah, actually it keeps the exit code because you are being thrown off by the `s we are grepping for the output of the command:
 xxd -p $FBASE_SNIPPET | tr -d '\n'

The output of grep isn't being piped to anything.